### PR TITLE
Feat/59 delete restore accommodation

### PR DESCRIPTION
**Pull request by Zach13131**
- confirmation modal on deleting
- ability to restore accommodation
- add the deleted label to the list of your accommodations
- make  getting user accommodation dynamic 
- fix some endpoints

### CODE REVIEW

File: src/pages/accommodations/AccommodationForm.tsx [hunk 1]
```diff
@@ -54,6 +60,8 @@ export default function AccommodationForm() {
     values: accommodation?.data,
   });
 
+  const isDeleted: boolean | undefined = accommodation?.data.isDeleted;
```
Comment (gitkoshelew) [inline src/pages/accommodations/AccommodationForm.tsx]: i think type should be only boolean !!accommodation?.data.isDeleted;

File: src/api/mutations/accommodations/useDeleteAccommodation.ts [hunk 2]
```diff
@@ -3,18 +3,27 @@ import { useMutation, useQueryClient } from '@tanstack/react-query';
...
   const queryClient = useQueryClient();
-  const { mutate } = useMutation({
+
+  const { mutate, isPending } = useMutation({
     mutationFn: (id: string) => {
       return httpClient.delete(ENDPOINTS.accommodation.deleteAccommodation(id));
     },
 
-    onSuccess: async () => {
+    onSuccess: async (_, accommodationId) => {
```
Comment (gitkoshelew) [inline src/api/mutations/accommodations/useDeleteAccommodation.ts:19]: handle Failure also here or inside the component

File: src/api/mutations/accommodations/useCreateAccommodation.ts [hunk 3]
```diff
@@ -14,6 +18,8 @@ export const useCreateAccommodation = () => {
 
     onSuccess: async () => {
```
Comment (gitkoshelew) [inline src/api/mutations/accommodations/useCreateAccommodation.ts:19]: handle Failure also here or inside the component

File: src/pages/accommodations/AccommodationForm.tsx [hunk 4]
```diff
@@ -82,11 +90,28 @@ export default function AccommodationForm() {
...
+
+  const handleRestore = useCallback(() => {
+    if (id) {
+      restoreAccommodation(id);
+    }
+  }, [id, restoreAccommodation]);
+
+  const [open, setOpen] = useState(false);
+
+  const handleClose = useCallback(() => setOpen(false), []);
```
Comment (fasbrothers) [inline src/pages/accommodations/AccommodationForm.tsx]: you can create one function instead of handleClose and handleClickOpen

**Review by gitkoshelew — APPROVED** [review]

**Review by fasbrothers — APPROVED** [review]
It looks good, but there is a merge conflict. It might be due to my changes, I changed the root directory from @ this to @src, as @adadychanka suggested.

**Pull request merged**